reagents reaction() now has more application method

phil235 opened the pull request:
- reagent reaction() now use more methods. Instead of just INGEST or TOUCH, we now have:
  - TOUCH (for splashing). It no longer transfer reagent by default, it's now only used for touch based effect like being flammable when touched by fuel or you and your stuff being acided when touched by acid). Fixes #11053 
  - INGEST (for injection, ingestion). No changes there.
  - PATCH (for patch application). It does no transfer of its own (but actual patches code do make a transfer) but is used for touch effect that ignore clothes and protection.
  - VAPOR (for smoke, foam and spray application (and blob attack). It does reagent transfer but it takes into account clothing protection
- Fixes a typo in humanoid/get_permeability_prot
- Fixes acid splashing on objects not destroying it. Fixes acid splashing on turf not destroying objects on the turf. Fixes #8959
- Changed damp rag smothering code a bit.
- Fixes blob sorium and dark matter vortex not working correctly. Fixes #11149
- Fixes blob expansion attacking its own blob mobs (blobbernaut,etc). Fixes #11208
- The effects of blob reagents from blob attack and blob smoke can now be mitigated by wearing permeable clothes such as hardsuits, medical clothes, masks. (the lower the permeability_coefficient the better). Changed blob smoke reagent amount from 5 to 10 to compensate so blob smoke effect aren't too affected by this.
- changed the volume argument of reaction_mob() to reac_volume to avoid confusion with the reagent var "volume".

phil235 committed:
reagent reaction() now use more methods. Instead of just INGEST or TOUCH, we now have INGEST (for injection, ingestion), TOUCH (for splashing), PATCH (for patch application and blob attack), and VAPOR (for smoke, foam and spray application).

- TOUCH no longer transfer reagent by default, it's now only used for touch based effect like being flammable when touched by fuel or you and your stuff being acided when touched by acid).

- PATCH does no transfer of its own (but actual patches code do make a transfer) but is used for touch effect that ignore clothes and protection (blob attacks, patches effect going through hardsuits)

- VAPOR does reagent transfer but it takes into account clothing protection

Fixes a typo in humanoid/get_permeability_prot
Fixes acid splashing on objects not destroying it. Fixes acid splashing on turf not destroying objects on the turf.

Changed damp rag smothering code a bit.

Blob Smoke now doesn't do anything.

GunHog commented:
> Blob Smoke now doesn't do anything (the blob reagents have no code for method == VAPOR).

Why would you allow that to be intended? Why not just add the proper code for the blob reagents so that they work properly? If you meant this as a nerf, the way you are doing it makes it seem as if you are introducing a bug.

Ergovisavi commented:
Shouldn't foam be TOUCH and smoke be VAPOR? Otherwise they're the same shit.

phil235 commented:
@GunHog It is an intended nerf, but if people want to keep blob smoke effect all I have to do is change `if(method ==PATCH)` to`if(method == PATCH || method == VAPOR)`

@Ergovisavi This PR doesn't change anything for foam or smoke, it's just a name change. Sure the VAPOR name isn't exactly a great fit for foam but the foam effects are unchanged.

Iamgoofball commented:
@Ergovisavi Also, foam needs to be able to transfer in reagents as well.

optimumtact commented on code/modules/detectivework/footprints_and_rag.dm:
```diff
@@ -22,14 +22,17 @@
-			reagents.clear_reagents()
+	if(iscarbon(A) && A.reagents && reagents.total_volume)
+		var/mob/living/carbon/C = A
+		if(user.a_intent == "harm" && !C.is_mouth_covered())
```
are there defines for intent type?

phil235 replied:
no, I don't think there is.

optimumtact replied:
rest in peace, carry on then :+1:

freerealestate commented:
If you're going to nerf blob, why not change it so its attacks don't ignore protection instead? Its attacks ignoring protection doesn't really make sense.

phil235 commented:
@freerealestate It's a question of balance. Do we really want every crewmember wearing a hardsuit or biosuit to be nearly immune to blob attacks?

freerealestate commented:
Nope, but I don't think it should their attacks should ignore protection
entirely.

On 12 August 2015 at 22:19, phil235 <notifications@github.com> wrote:

> @freerealestate <https://github.com/freerealestate> It's a question of
> balance. Do we really want every crewmember wearing a hardsuit or biosuit
> to be nearly immune to blob attacks?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/11119#issuecomment-130249629>
> .
>

phil235 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into VaporizeMethodReaction

Conflicts:
	code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm

phil235 commented:
don't merge this yet, still got something to add.

phil235 committed:
Readds blob smoke reagent effect.
You can now partially (max is 50%) protect yourself again blob reagent effects (blob smoke and blob attack). Although some effects are unaffected (explosion probability, vortex power, adding fire_stacks).
Changes the "volume" argument used in reaction_mob() to "reac_volume" to avoid confusion with the "volume" variable.
Buffed blob smoke reagent amount from 5 to 10 to counterbalance the nerf from clothes protection.
Blob direct attack now uses the VAPOR method.

phil235 commented:
Alright, I added the fixes of #11210 to this PR. I've also readded the blob reagent effects to blob smoke.
Also added blob effect protection with permeable clothes, see the OP.
